### PR TITLE
[scudo] Avoid accessing inaccessible pages in unmap() in secondary

### PR DESCRIPTION
**Pull request by ChiaHungDuan**

### CODE REVIEW

**Comment by ChiaHungDuan** [comment]
@Caslyn , could you help verify if this fixes the problem on Fuchsia?

File: compiler-rt/lib/scudo/standalone/secondary.h [hunk 1]
```diff
@@ -823,7 +823,11 @@ void MapAllocator<Config>::deallocate(const Options &Options, void *Ptr)
     Cache.store(Options, H->CommitBase, H->CommitSize,
                 reinterpret_cast<uptr>(H + 1), H->MemMap);
   } else {
-    unmap(H->MemMap);
+    // Note that the `H->MapMap` is stored on the pages managed by itself. Take
```
Comment (cferris1000) [inline compiler-rt/lib/scudo/standalone/secondary.h:826]: MemMap

**Review by cferris1000 — COMMENTED** [review]
Comment nit.

**Review by Caslyn — APPROVED** [review]
Confirming this clears up issues in Fuchsia - thanks for this fix!

File: compiler-rt/lib/scudo/standalone/secondary.h (hunk 1 above)
Comment (ChiaHungDuan) [reply compiler-rt/lib/scudo/standalone/secondary.h]: Done

**Pull request merged**